### PR TITLE
Upgrade to Hapi 9.x.x

### PR DESCRIPTION
**Pull request by asaf**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,15 +19,16 @@
     "items": "1.x.x",
     "joi": "6.x.x",
     "jsonic": "0.1.x",
-    "seneca": "0.6.x"
+    "seneca": "0.6.x",
+    "vision": "^3.0.0"
   },
   "peerDependencies": {
-    "hapi": "^8.5.x"
+    "hapi": "^9.x.x"
```
Comment (hueniverse) [inline package.json]: This needs to be `>=8.5.0` now.

Comment (asaf) [inline package.json]: Agreed =)

**Comment by hueniverse** [comment]
Probably need to use `server.dependency()` on vision.

**Comment by asaf** [comment]
@hueniverse Instead of server.register in the unit tests? or within the plugin code itself? IMO plugin itself should not be dependent on vision as many people don't use Hapi for server side rendering.

**Comment by hueniverse** [comment]
The plugin already depends on vision. If we want to allow using it without it we need to make that an option.

**Comment by blainsmith** [comment]
This would be awesome to support Hapi 9 and above. I have a pre-existing seneca framework I want to use Hapi with and long with other Hapi plugins like Lout, etc.

**Comment by hueniverse** [comment]
hapi 9.x is already supported, just need to make sure you load vision yourself first.

**Comment by asaf** [comment]
@hueniverse "The plugin already depends on vision", it's currently commented out
https://github.com/hapijs/chairo/blob/master/lib/index.js#L33

I don't think this is related directly to this change that bump the version of Hapi peer dependency,

Can we merge this to free us from the 8.x hard requirement and handle the vision dependency in another issue?


Thanks

**Comment by Marsup** [comment]
It does implicitly depend on vision because of [this line](https://github.com/hapijs/chairo/blob/1993031c3230fc531f38699f3c511a722dc8da2b/lib/index.js#L152).

**Comment by asaf** [comment]
@Marsup That's a bug then, either vision is optional or dependency should explicitly defined, we can easily make it optional via configuration and avoid registering the `compose` if vision is not enabled,

Either way, I don't see how this is related to this PR.

**Comment by Marsup** [comment]
It is **not** currently optional. Hapi came with vision preloaded before, that's not the case anymore in 9, you need to depend on it.

**Comment by asaf** [comment]
@Marsup Many users just don't use server side rendering, I don't see why I need this into my stack just because I want to use Chairo, that was the whole point of extracting it out from the core from the first place, I do think this should be optional, up to @hueniverse to decide.

**Comment by Marsup** [comment]
You'd have to drop one of the features (compose) if you don't want vision. So you either lack dependency on vision or an option to disable compose. This can't be merged as is.

**Comment by hueniverse** [comment]
Hmm, we are missing an optional dependency feature...

**Comment by asaf** [comment]
@hueniverse Do you want me to add peer dependency on vision for now or prefer to keep this open until we'll come up with a better optional dependency support?

**Comment by hueniverse** [comment]
Is this blocking you from moving to 9.x?

**Comment by asaf** [comment]
@hueniverse Peer dependency causes failure (`"hapi": "^8.5.x"`), for the meantime I'm using a fork, but i'll be _hapi_ if the official branch will support 9.x in the near time.

Thanks.

**Comment by gergoerdosi** [comment]
Any news on this PR? Chairo can't even be installed with hapi 10:

```
npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "hapi" "chairo"
npm ERR! node v4.0.0
npm ERR! npm  v2.14.2
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package hapi@10.0.0 does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer chairo@0.0.1 wants hapi@^8.5.x
```

**Comment by geek** [comment]
@gergoerdosi this is now merged.  I am going to do a final review before publishing, which should happen today.

**Comment by geek** [comment]
Thanks, @hueniverse

**Pull request merged**